### PR TITLE
Vuetify published histories

### PR DESCRIPTION
**Pull request by itisAliRH**
Followup #14907 and #14893
This pr vuetified the `Published Histories` component.

image

The current API has some limitations:
- filtering by `annotation` is case sensitive and does not return the right result (client always sends queries in lowercase)
- filtering by `owner/username` is not available
- sort by `owner/username` is not available

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [x] Instructions for manual testing are as follows:
  1. Share a history
  2. Go to `/histories/list_published`

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: client/src/components/History/services.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import axios from "axios";
+import { prependPath } from "utils/redirect";
+import { rethrowSimple } from "utils/simple-error";
+
+export class Services {
+    async getPublishedHistories(options = {}) {
```
Comment (guerler) [inline client/src/components/History/services.js]: This can be exported directly as a function without the `Services` class wrapper. Since the class is stateless.

File: client/src/components/History/HistoryPublishedList.vue [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+import UtcDate from "components/UtcDate";
+import Tags from "components/Common/Tags";
+import { safePath } from "utils/redirect";
+import { computed, ref, watch } from "vue";
+import Heading from "components/Common/Heading";
+import LoadingSpan from "components/LoadingSpan";
+import DebouncedInput from "components/DebouncedInput";
+import { getPublishedHistories, updateTags } from "./services";
+import { getFilters, toAlias } from "store/historyStore/model/filtering";
+import { getFilterText } from "./CurrentHistory/HistoryFilters/filterConversion";
```
Comment (guerler) [inline client/src/components/History/HistoryPublishedList.vue]: We should move this file to a shared location maybe `utils/filter.js`.

File: client/src/components/History/HistoryPublishedList.vue [hunk 3]
```diff
@@ -0,0 +1,233 @@
+<script setup>
+import UtcDate from "components/UtcDate";
+import Tags from "components/Common/Tags";
+import { safePath } from "utils/redirect";
+import { computed, ref, watch } from "vue";
+import Heading from "components/Common/Heading";
+import LoadingSpan from "components/LoadingSpan";
+import DebouncedInput from "components/DebouncedInput";
+import { getPublishedHistories, updateTags } from "./services";
+import { getFilters, toAlias } from "store/historyStore/model/filtering";
```
Comment (guerler) [inline client/src/components/History/HistoryPublishedList.vue]: We should refactor this so that other published components can use these `filterOperations` without having to import history store internals. I think we should move parts of this functionality to a shared directory and consolidate it in `utils/filter.js`. `validFilters` is probably the only function which should remain in the history context, since it is history specific.

File: client/src/components/History/services.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import axios from "axios";
+import { safePath } from "utils/redirect";
+import { rethrowSimple } from "utils/simple-error";
+import { getQueryDict } from "store/historyStore/model/filtering";
+
+const getQueryString = (filterText) => {
```
Comment (guerler) [inline client/src/components/History/services.js]: This should also be moved to `filter.js`, its likely a duplicate and is already used in `components/History/model/queries.js` and `store/historyStore/historyItemsStore.js`.

**Comment by itisAliRH** [comment]
@guerler Thank you for the review :)
All `filtering` related methods/vars are moved to `utils/filterConversation.js` and accessible for all components.

**Comment by guerler** [comment]
Talked about this, and our plan is right now to get this in pretty much as is and then follow-up by improving the modularity and interface for filter helpers such that they can be used not only to filter histories but also other objects such as e.g. workflows etc.

File: client/src/utils/filterConversion.js [hunk 5]
```diff
@@ -10,41 +10,53 @@
...
 /** Converts user input to backend compatible date. */
-function toDate(value) {
+const toDate = (value) => {
     return Date.parse(value) / 1000;
-}
+};
 
 /** Converts user input for case-insensitive filtering. */
-function toLower(value) {
+export const toLower = (value) => {
```
Comment (guerler) [inline client/src/utils/filterConversion.js]: Talked about this, we intend to restore the `function NAME() {}` notation here. More details will follow separately. We intend to add this and other conventions to the docs for clarificiation but thats not part of this PR.

**Review by guerler — APPROVED** [review]
Looks good to me. Thank you so much.

**Comment by itisAliRH** [comment]
The failed test is unrelated to this pr :)
